api/v1alpha1/backend_types.go

```diff
@@ -0,0 +1,156 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// KindBackend is the name of the Backend kind.
+	KindBackend = "Backend"
+)
+
+// +kubebuilder:validation:Enum=FQDN;UDS;IPv4;IPv6
+// +notImplementedHide
+type AddressType string
+
+const (
+	// AddressTypeFQDN defines the RFC-1123 compliant fully qualified domain name address type.
+	AddressTypeFQDN ProtocolType = "FQDN"
+	// AddressTypeUDS defines the unix domain socket address type.
+	AddressTypeUDS ProtocolType = "UDS"
+	// AddressTypeIPv4 defines the IPv4 address type.
+	AddressTypeIPv4 ProtocolType = "IPv4"
+	// AddressTypeIPv6 defines the IPv4 address type.
+	AddressTypeIPv6 ProtocolType = "IPv6"
```

Comment:
can we support IPv4 and IPv6 at the same time?

Reply:
I couldn't find anything in the envoy docs to suggest that it's not possible. Judging by this example: https://istio.io/latest/blog/2023/experimental-dual-stack/, istio uses both IPv4 and IPv6 endpoints in the same cluster for dualstack support, as seen here:

```
ENDPOINT                 STATUS      OUTLIER CHECK     CLUSTER
[...]
10.244.0.26:9000         HEALTHY     OK                outbound|9000||tcp-echo.dual-stack.svc.cluster.local
fd00:10:244::1a:9000     HEALTHY     OK                outbound|9000||tcp-echo.dual-stack.svc.cluster.local
[...]
```

Having said that, I think that we can avoid IPv6 support in this API for the time being, and maybe address this at a later stage, together with #184.